osidb/sync_manager.py

```diff
@@ -85,8 +99,31 @@ def schedule(cls, sync_id, *args, schedule_options=None, **kwargs):
         if schedule_options is None:
             schedule_options = {}
 
-        cls.objects.get_or_create(sync_id=sync_id)
-        cls.objects.filter(sync_id=sync_id).update(last_scheduled_dt=timezone.now())
+        now = timezone.now()
+        created = cls.objects.get_or_create(sync_id=sync_id)[1]
+        cls.objects.filter(sync_id=sync_id).update(last_scheduled_dt=now)
+
+        if not created and cls.MODE == cls.SyncManagerMode.EXCLUSIVE:
+            # If the task is already scheduled, skip
+            if cls.is_scheduled(sync_id):
+                logger.info(
+                    f"{cls.__name__} {sync_id}: Task already scheduled, skipping"
+                )
+                return
+
+            # If the task is already in progress, schedule it with a countdown
+            if cls.is_in_progress(sync_id):
+                logger.info(
+                    f"{cls.__name__} {sync_id}: Task already in progress"
+                    f", postponing for {cls.COUNTDOWN} seconds"
+                )
+                if "countdown" not in schedule_options:
+                    schedule_options["countdown"] = cls.COUNTDOWN
```

Comment:
This could produce log message that the task will be postponed with `cls.COUNTDOWN` value, but different value would be applied instead if `schedule_options` would already contain `countdown`